pkg/code-generator/model/project.go

```diff
@@ -65,10 +67,15 @@ type Resource struct {
 }
 
 type Field struct {
-	Name     string
-	TypeName string
-	IsOneof  bool // we ignore oneof fields in test generation
-	Original *descriptor.FieldDescriptorProto
+	Name        string
+	TypeName    string
+	IsOneof     bool // we ignore oneof fields in test generation
+	SkipHashing bool // skip this field when hashing the resource
+	Original    *descriptor.FieldDescriptorProto
+}
+
+type Oneof struct {
+	Name string
```

Comment:
because we are hashing individual fields rather than the whole object, we must account for `Oneof` fields